### ISSUE TITLE
Test stubs aren't consumed by the connection.

### ISSUE THREAD

**Issue by lucasmazza** [lostisland/hurley#18]
I don't know if this is a desired behavior of the Test connection, but it is one that made a lot of sense when working with Faraday. When defining multiple stubs for the same URI, only the first one gets used by the connection and the subsequent responses aren't used, even when the client perform multiple requests

```ruby
connection.get('/test') { [200, {}, 'foo'] }
connection.get('/test') { [200, {}, 'bar'] }

client.get('/test').body # => 'foo'
client.get('/test').body # => Expected to be 'bar', but it's 'foo'.
```

Is this the expected behavior of some oversight when porting the API from Faraday?

**Comment by kkirsche** [comment]
I can confirm that this happens to me as well.

**Comment by technoweenie** [comment]
Honestly I forgot this was a thing in Faraday :/  I'm in favor of adding it.

**Comment by kkirsche** [comment]
@technoweenie the actual deletion of the test stubs is easy (`next if h.run?;` in `test.rb#Hurley::Test.call45`). The problem is I can't easily comprehend and fix your client tests as they are batched in such a way my lack of familiarity makes it not worth the time to learn all the different design intricities

**Comment by technoweenie** [comment]
I think the problem with the other tests is they rely on calling the same stub multiple times. Your proposed fix breaks that by skipping handlers even if the test isn't relying on multiple stubbed responses.

I think we can just make this explicit.

```ruby
# stub never expires
connection.get('/test') { [200, {}, 'foo'] }

# this stub expires
connection.get('/test', expires: true) { [200, {}, 'foo'] }

# this stub expires after the second one is created
connection.get('/test') { [200, {}, 'foo'] }
connection.get('/test') { [200, {}, 'bar'] }

# the second /test makes the first one expire
connection.get('/test', expires: false) { [200, {}, 'foo'] }
connection.get('/test') { [200, {}, 'bar'] }
```

**Comment by technoweenie** [comment]
Should be fixed in #26.
